apps/server/src/controllers/ontimeController.ts

```diff
@@ -489,7 +490,9 @@ export const listProjects: RequestHandler = async (_, res: Response<ProjectFileL
   try {
     const fileList = await getProjectFiles();
 
-    const lastLoadedProject = JSON.parse(fs.readFileSync(lastLoadedProjectConfigPath, 'utf8')).lastLoadedProject;
+    const lastLoadedProject = removeFileExtension(
+      JSON.parse(fs.readFileSync(lastLoadedProjectConfigPath, 'utf8')).lastLoadedProject,
+    );
```

Comment:
suggestion: could we wrap this into a few steps?

I think this style of function composition tends do impact the ability to read and debug without that many advantages. At least not to this style.
If we were to have some piping system like

```ts

const lastLoadedProject = pipe(
  readFile,
  parse,
  getLastLoadedProject,
  ["lastLoadedProjectConfigPath"]
)
```

There could be some value. but that is not a pattern we use here or have artifices for

Reply:
agreed